src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,168 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> tail;
+    private Node<T> head;
+    private int size;
+
     @Override
     public void add(T value) {
+        linkTail(value);
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexToAdd(index);
+        if (index == size) {
+            linkTail(value);
+        } else if (index == 0) {
+            linkHead(value);
+        } else {
+            Node<T> nodeBeforeIndex = getNodeByIndex(index - 1);
+            Node<T> newNode = new Node<>(nodeBeforeIndex, value, nodeBeforeIndex.next);
+            nodeBeforeIndex.next.prev = newNode;
+            nodeBeforeIndex.next = newNode;
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element : list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> node = getNodeByIndex(index);
```

Comment:
This is code repeating. You always check the index before getting node by index. What you should do is move the check logic INSIDE the get method.